src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,294 @@
+        {
+            if (declaredNamespaceParts.IsDefault || targetNamespaceParts.IsDefault || !(root is CompilationUnitSyntax compilationUnit))
+            {
+                return root;
+            }
+
+            // Move everything from global namespace to a namespace declaration
+            if (declaredNamespaceParts.Length == 1 && declaredNamespaceParts[0].Length == 0)
+            {
+                var targetNamespaceDecl = SyntaxFactory.NamespaceDeclaration(
+                    name: CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithAdditionalAnnotations(WarningAnnotation),
+                    externs: default, 
+                    usings: default,
+                    members: compilationUnit.Members);
+                return compilationUnit.WithMembers(new SyntaxList<MemberDeclarationSyntax>(targetNamespaceDecl))
+                    .WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // We should have a single member which is a namespace declaration in this compilation unit.
+            var namespaceDeclaration = root.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+
+            // Move everything to global namespace
+            if (targetNamespaceParts.Length == 1 && targetNamespaceParts[0].Length == 0)
+            {
+                var triviaFromNamespaceDecl = GetOpeningAndClosingTriviaOfNamespaceDeclaration(namespaceDeclaration);
+
+                var members = namespaceDeclaration.Members;
+                var eofToken = compilationUnit.EndOfFileToken
+                    .WithAdditionalAnnotations(WarningAnnotation);
+
+                // Try to preserve trivia from original namesapce declaration.
+                // If there's any member inside the declaration, we attach them to the 
+                // first and last member, otherwise, simply attach all to the EOF token.
+                if (members.Count > 0)
+                {
+                    var first = members.First();
+                    var firstWithTrivia = first.WithPrependedLeadingTrivia(triviaFromNamespaceDecl.openingTrivia);
+                    members = members.Replace(first, firstWithTrivia);
+
+                    var last = members.Last();
+                    var lastWithTrivia = last.WithAppendedTrailingTrivia(triviaFromNamespaceDecl.closingTrivia);
+                    members = members.Replace(last, lastWithTrivia);
+                }
+                else
+                {
+                    eofToken = eofToken.WithPrependedLeadingTrivia(
+                        triviaFromNamespaceDecl.openingTrivia.Concat(triviaFromNamespaceDecl.closingTrivia));
+                }
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
```

Comment:
we should comment that this can lead to a break in semantics (and have tests that show that).  When a using is pulled out of a namespace, it changes meaning.  i.e. you might have this:

```c#
namespace A.B.C
{
   using D.E.F;
}
```

That using is looked up first within the context of `A.B.C`.  If you move the 'using' to the outside, it may fail to resolve.  I think that's ok, but we need to call out that we recognize that limitation.